src/Microsoft.OData.Core/JsonLight/ODataJsonLightDeserializer.cs

```diff
@@ -908,6 +1099,369 @@ private void ReadODataOrCustomInstanceAnnotationValue(string annotatedPropertyNa
                 this.AssertJsonCondition(JsonNodeType.Property, JsonNodeType.EndObject);
             }
 
+            return null;
+        }
+        /// <summary>
+        /// If <paramref name="annotationName"/> is under the odata namespace but is not known to ODataLib, move the JSON reader forward to skip the
+        /// annotation name and value then return true; return false otherwise.
+        /// </summary>
+        /// <remarks>
+        /// The unknown odata annotation is skipped so that when this version of the reader reads a resource set produced by a future version of ODataLib
+        /// that contains an odata annotation that is not recognized on this version, we would simply ignore the annotation rather than failing.
+        /// Note that when we add new odata annotations that cannot be skipped, we would bump the protocol version.
+        /// </remarks>
+        /// <param name="annotationName">The annotation name in question.</param>
+        /// <returns>A task that represents the asynchronous read operation.
+        /// The value of the TResult parameter contains true if the annotation name and value is skipped; otherwise false.</returns>
+#if NETSTANDARD2_0
+        private async ValueTask<Tuple<bool, object>> SkipOverUnknownODataAnnotationAsync(string annotationName)
+#else
+        private async Task<Tuple<bool, object>> SkipOverUnknownODataAnnotationAsync(string annotationName)
+#endif
+        {
+            Debug.Assert(!string.IsNullOrEmpty(annotationName), "!string.IsNullOrEmpty(annotationName)");
+            this.AssertJsonCondition(JsonNodeType.Property);
+
+            object annotationValue = null;
+            bool isUnknownODataAnnotationName = false;
+
+            if (ODataAnnotationNames.IsUnknownODataAnnotationName(annotationName))
+            {
+                annotationValue = await ReadODataOrCustomInstanceAnnotationValueAsync(annotationName)
+                    .ConfigureAwait(false);
+                isUnknownODataAnnotationName = true;
+            }
+
+            return Tuple.Create(isUnknownODataAnnotationName, annotationValue);
+        }
+
+        /// <summary>
+        /// Asynchronously reads "odata." or custom instance annotation's value.
+        /// </summary>
+        /// <param name="annotationName">The annotation name.</param>
+        /// <returns>The annotation value.</returns>
+        /// <returns>A task that represents the asynchronous read operation.
+        /// The value of the TResult parameter contains the annotation value.</returns>
+#if NETSTANDARD2_0
+        private async ValueTask<object> ReadODataOrCustomInstanceAnnotationValueAsync(string annotationName)

```

Comment:
Out of curiosity, why does this return a `ValueTask`, all code paths seem to be async? Or do async paths also return `ValueTask`s?

Reply:
For now removed the use of ValueTask with the plan being to going in after and applying that where actual value can be derived